keras_nlp/layers/transformer_decoder.py

```diff
@@ -212,35 +212,57 @@ def call(
+        self_attention_cache_update_index=0,
+        cross_attention_cache=None,
+        cross_attention_cache_update_index=0,
     ):
         """Forward pass of the TransformerDecoder.
 
         Args:
             decoder_sequence: a Tensor. The decoder input sequence.
             encoder_sequence: a Tensor. The encoder input sequence. For decoder
-                only models (like GPT2), this should be left None. Once the
+                only models (like GPT2), this should be left `None`. Once the
                 model is called once without an encoder_sequence, you cannot
                 call it again with encoder_sequence.
             decoder_padding_mask: a boolean Tensor, the padding mask of decoder
-                sequence, must of shape [batch_size, decoder_sequence_length].
+                sequence, must be of shape
+                `[batch_size, decoder_sequence_length]`.
             decoder_attention_mask: a boolean Tensor. Customized decoder
-                sequence mask, must of shape
-                [batch_size, decoder_sequence_length, decoder_sequence_length].
+                sequence mask, must be of shape
+                `[batch_size, decoder_sequence_length, decoder_sequence_length]`.
             encoder_padding_mask: a boolean Tensor, the padding mask of encoder
-                sequence, must of shape [batch_size, encoder_sequence_length].
+                sequence, must of shape `[batch_size, encoder_sequence_length]`.
             encoder_attention_mask: a boolean Tensor. Customized encoder
                 sequence mask, must of shape
-                [batch_size, encoder_sequence_length, encoder_sequence_length].
-            cache: a dense float Tensor. The cache of key/value of leading
-                tokens. `cache` is of shape [B, 2, max_seq_len, num_heads,
-                key_dims].
-            cache_index: a int or int Tensor, the index of the current token
-                being processed.
+                `[batch_size, encoder_sequence_length, encoder_sequence_length]`.
+            self_attention_cache: a dense float Tensor. The cache of key/value
+                of leading tokens in the self-attention layer. Of shape
+                `[batch_size, 2, max_seq_len, num_heads, key_dims]`.
+            self_attention_cache_update_index: an int or int Tensor, the index
+                of the current token (in `decoder_sequence`) being processed.
+                Both `self_attention_cache` and
+                `self_attention_cache_update_index` have to be non-`None` for
+                the cache to be updated.
+            cross_attention_cache: a dense float Tensor. The cache of
+                key/value pairs of encoder outputs. `cache` is of shape
+                `[batch_size, 2, S, num_heads, key_dims]`.
+            cross_attention_cache_update_index: an int or int Tensor, the index
+                of the current token (in `encoder_sequence`) being processed.
```

Comment:
the index at which to update the `cross_attention_cache`. As the `cross_attention_cache` only need to be computed once, this is usually either `0` (update the whole cache) or `None` (no updates needed).